### ISSUE TITLE
Known spell lists resets on player death

### ISSUE THREAD

**Issue by flamechair** [Electroblob77/Wizardry#69]
Minecraft version: 1.12.2
SpongeAPI: 7.1.0-SNAPSHOT-6cd2dc3
SpongeForge: 1.12.2-2705-7.1.0-BETA-3399
Minecraft Forge: 14.23.4.2705
Wizardry version: 4.1.2
Environment: Server

Issue details: 
1. Learn some spells
2. Die (I tested via /kill)
3. Spells becomes undiscovered (both in wands and spellbooks)

Other mods involved:
Not sure which ones are relevant, so here is the list of them:
```
Mods (95): Minecraft, Minecraft Coder Pack, Forge Mod Loader, Minecraft Forge, SpongeAPI,
SpongeForge, AI Improvements, AbyssalCraft, AbyssalCraft Integration, Aroma1997Core, AromaBackup, 
AromaBackuprecovery, Astral Sorcery, Bad Wither No Cookie! Reloaded, Baubles, BiblioCraft, Biomes O' 
Plenty, Blood Magic: Alchemical Wizardry, Bookshelf, Botania, Botania Tweaks, Botania Tweaks Core, 
Chameleon, ChickenChunks, Chisel, CoFH Core, CoFH World, CodeChicken Lib, CookieCore, Cyclops 
Core, DMT, Dark Utilities, Defiled Lands, Dimensional Doors, Disenchanter, Dungeon Mods, Dynamic 
Surroundings, Electroblob's Wizardry, EvilCraft, EvilCraft-Compat, Fence Jumper, FoamFix, FoamFixCore, 
Forgiving Void, Guide-API, HellBound, Howling Moon, Ice and Fire, InControl, InstantUnify, Inventory 
Pets, Inventory Tweaks, Iron Chest, Just Enough Items, LLibrary, LagGoggles, LibEx, Magic Feather, 
McJtyLib, Mod Control, Nether Foundation, NetherBound, NetherEx, NetherPortalFix, No Mob Spawning 
on Trees, Not Enough Items, OpenBlocks, OpenModsLib, OpenModsLib Core, Progressive Bosses, 
RFTools, Raiders, Redstone Flux, Roguelike Dungeons, Storage Drawers, TOP Addons, TeamLapen Library, 
Thaumcraft, The One Probe, The Twilight Forest, Thermal Cultivation, Thermal Dynamics, Thermal 
Expansion, Thermal Foundation, TickProfiler, Tree Chopper, UniDict, Unlimited Chisel Works, Vampirism, 
VampirismIntegrations, VanillaFix, WanionLib, Waystones, WorldEdit, Worley's Caves
```

I've verified that in SP this issue does not happen

**Comment by flamechair** [comment]
Leaving the server and logging back in restores the list of known spells correctly

**Comment by Electroblob77** [comment]
Ok, so it's a syncing issue then. That should be a pretty easy fix, thanks for reporting it.

**Comment by Electroblob77** [comment]
Fixed in the 4.2.0 release.
